App.js

```diff
@@ -31,61 +35,66 @@ class App extends Component {
             },
             [navKeys.AUTH]: {
-                screen: AuthScreen,
-                navigationOptions: {
-                    gesturesEnabled: false,
-                }
+                screen: AuthScreen
             },
             [navKeys.MAIN]: {
                 screen: TabNavigator({
                     [navKeys.SEARCH]: {
-                        screen: SearchScreen,
-                        navigationOptions: {
-                            gesturesEnabled: false,
-                        }
+                        screen: SearchScreen
                     },
                     [navKeys.FAVORITES]: {
-                        screen: FavoritesScreen,
-                        navigationOptions: {
-                            gesturesEnabled: false,
-                        }
+                        screen: FavoritesScreen
                     },
                     [navKeys.ADD_SITE]: {
-                        screen: AddSiteScreen,
-                        navigationOptions: {
-                            gesturesEnabled: false,
-                        }
+                        screen: AddSiteScreen
                     },
                     [navKeys.MORE]: {
-                        screen: MoreScreen,
-                        navigationOptions: {
-                            gesturesEnabled: false,
-                        }
+                        screen: MoreScreen
                     }
                 })
             },
             [navKeys.FILTER]: {
                 screen: FilterScreen,
                 navigationOptions: {
-                    tabBarVisible: false,
-                    gesturesEnabled: false,
+                    tabBarVisible: false
                 }
             },
             [navKeys.ADD_SITE_FORM]: {
                 screen: AddSiteFormScreen,
                 navigationOptions: {
-                    tabBarVisible: false,
-                    gesturesEnabled: false,
+                    tabBarVisible: false
+                }
+            },
+            [navKeys.SITE_DETAIL]: {
+                screen: SiteDetailScreen,
+                navigationOptions: {
+                    tabBarVisible: false
+                }
+            },
+            [navKeys.SITE_DETAIL_MAP_VIEW]: {
+                screen: SiteDetailMapViewScreen,
+                navigationOptions: {
+                    tabBarVisible: false
                 }
             }
         }, {
-            lazy: true
+            navigationOptions: {
+                gesturesEnabled: false,
+                headerTitleStyle: {
+                    color:'white'
+                },
+                headerStyle: {
+                    backgroundColor: blueGreenNav
+                },
+                headerBackTitle: common.back,
+                headerBackTitleStyle: {
+                    color: linkColorBlue
+                },
+                headerTintColor: linkColorBlue
+            }
```

Comment:
@amgoodman5 I figured out how to do app-wide default styles for the navigation bar. We can drop any defaults here, and if needed we can do specific overrides in each of the different components. Further up this file you'll see I removed all the repetitive code and just dropped it all here. Also, in the individual components below you'll see I removed the repetitive code. I

Reply:
<3